kernel/src/process.rs

```diff
@@ -330,15 +330,56 @@ impl<'a> Process<'a> {
         if data_len.count_ones() != 1 {
             panic!("Tock MPU does not currently handle complex region sizes");
         }
-        let data_region_len = math::log_base_two(data_len as u32);
+        let data_region_len = math::PowerOfTwo::from(data_len as u32);
 
+        // Text segment read/execute (no write)
         let text_start = self.text.as_ptr() as usize;
         let text_len = self.text.len();
         if text_len.count_ones() != 1 {
             panic!("Tock MPU does not currently handle complex region sizes");
         }
-        let text_region_len = math::log_base_two(text_len as u32);
 
+        // Invariant: text_len is a power of two
+        if text_start % text_len != 0 {
+            // Text length not aligned to text start
+            let div = text_start % text_len;
```

Comment:
consider renaming `div` to `subregion_size`, I got confused the first time I walked through this